### ISSUE TITLE
False negative for `Style/ReturnNilInPredicateMethodDefinition` when last expression returns conditionally

### ISSUE THREAD

**Issue by Earlopain** [rubocop/rubocop#13064]
## Expected behavior

Offense for the following:

```rb
def foo?
  true if bar?
end
```

This returns nil implicitly when the condition is not met. Compantion to #13063.

Currently the cop seems to only focus on explicit returns but I think these types of things are in scope for it.

**Comment by koic** [comment]
Same as #13063.

`Style/ReturnNilInPredicateMethodDefinition` is intended to check only the `return` keyword. So, this proposal doesn't align with the role of `Style/ReturnNilInPredicateMethodDefinition`. Checking the last evaluated value should be the role of a different cop.
